src/category_theory/sites/grothendieck.lean

```diff
@@ -358,6 +359,143 @@ def atomic (hro : right_ore_condition C) : grothendieck_topology C :=
     exact ⟨_, _, hg⟩,
   end }
 
+/-- `J.cover X` denotes the poset of covers of `X` with respect to the
+Grothendieck topology `J`. -/
+@[derive preorder]
+def cover (X : C) := { S : sieve X // S ∈ J X }
+
+namespace cover
+
+variables {J}
+
+instance : has_coe (J.cover X) (sieve X) := ⟨λ S, S.1⟩
+
+instance : has_coe_to_fun (J.cover X) (λ S, Π ⦃Y⦄ (f : Y ⟶ X), Prop) :=
+⟨λ S Y f, (S : sieve X) f⟩
+
+@[simp]
+lemma coe_fun_coe (S : J.cover X) (f : Y ⟶ X) : (S : sieve X) f = S f := rfl
+
+lemma condition (S : J.cover X) : (S : sieve X) ∈ J X := S.2
+
+@[ext]
+lemma ext (S T : J.cover X) (h : ∀ ⦃Y⦄ (f : Y ⟶ X), S f ↔ T f) : S = T :=
+subtype.ext $ sieve.ext h
+
+instance : semilattice_inf_top (J.cover X) :=
+{ inf := λ S T, ⟨S ⊓ T, J.intersection_covering S.condition T.condition⟩,
+  le_antisymm := λ S T h1 h2, ext _ _ $ λ Y f, ⟨h1 _, h2 _⟩,
+  inf_le_left := λ S T Y f hf, hf.1,
+  inf_le_right := λ S T Y f hf, hf.2,
+  le_inf := λ S T W h1 h2 Y f h, ⟨h1 _ h, h2 _ h⟩,
+  top := ⟨⊤, J.top_mem _⟩,
+  le_top := λ S Y f h, by tauto,
+  ..(infer_instance : preorder _) }
+
+instance : inhabited (J.cover X) := ⟨⊤⟩
+
+/-- An auxiliary structure, used to define `S.index` in `plus.lean`. -/
```

Comment:
These names `L` and `R` are very "unreadable". If you only plan to use this in `plus.lean`, I would suggest moving them there, and marking them as implementation detail. If you want to expose them to others, maybe we can cook up a more descriptive name + docstring?

Reply:
I think I want to keep these definitions in this file, since they will be used is `sheaf.lean` in my next PR. (Actually, I just shuffled some things around and moved the `cover.index` definition from `plus.lean` to `grothendieck.lean` for the same reason). 

The names `L` and `R` are meant to match the fields in `cover.index` which in turn match the fields in `multicospan_index`, so that's one possible argument for keeping them. Although I agree they're not very descriptive... I'll try to come up with a better name.

Reply:
Okay! I renamed `S.L -> S.arrow` and `S.R -> S.relation` for `S : J.cover X`.

Reply:
Great! I like this.